File: ontrack-extension-ldap/src/main/java/net/nemerosa/ontrack/extension/ldap/LDAPMapping.java
package net.nemerosa.ontrack.extension.ldap;

import lombok.Data;
import lombok.EqualsAndHashCode;
import net.nemerosa.ontrack.model.security.AccountGroup;
import net.nemerosa.ontrack.model.security.AccountGroupMapping;
import net.nemerosa.ontrack.model.structure.ID;

import java.beans.ConstructorProperties;

@EqualsAndHashCode(callSuper = false)
@Data
public class LDAPMapping extends AccountGroupMapping {

    @ConstructorProperties({"id", "type", "name", "group"})
    public LDAPMapping(ID id, String type, String name, AccountGroup group) {
        super(id, type, name, group);
    }

    public static LDAPMapping of(AccountGroupMapping mapping) {
        return new LDAPMapping(
                mapping.getId(),
                mapping.getType(),
                mapping.getName(),
                mapping.getGroup()
        );
    }

}

File: ontrack-extension-ldap/src/main/java/net/nemerosa/ontrack/extension/ldap/LDAPMapping.kt
package net.nemerosa.ontrack.extension.ldap

import net.nemerosa.ontrack.model.security.AccountGroup
import net.nemerosa.ontrack.model.security.AccountGroupMapping
import net.nemerosa.ontrack.model.structure.ID

class LDAPMapping(
        val id: ID,
        val type: String,
        val name: String,
        val group: AccountGroup
) {
    companion object {
        @JvmStatic
        fun of(mapping: AccountGroupMapping): LDAPMapping = LDAPMapping(
                mapping.id,
                mapping.type,
                mapping.name,
                mapping.group
        )
    }
}